### PR TITLE
add tests for KUDO generic operator addon

### PR DESCRIPTION
**Pull request by zmalik**
adds tests for the kudo generic application operator addon: https://github.com/mesosphere/kubeaddons-enterprise/pull/49

### CODE REVIEW

File: tests/kubeaddons-enterprise/addons/generic-application-operator/0.x/generic-operator-install-0-x/00-install.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+apiVersion: kudo.dev/v1beta1
+kind: TestStep
+commands:
+  - command: kubectl apply -f ${KUBEADDONS_ENTERPRISE_ABS_PATH}/addons/generic-application-operator/0.x/generic-application-operator.yaml
```
Comment (porridge) [inline tests/kubeaddons-enterprise/addons/generic-application-operator/0.x/generic-operator-install-0-x/00-install.yaml:4]: Why don't we just symlink the file here?

File: tests/kubeaddons-enterprise/addons/generic-application-operator/0.x/generic-operator-install-0-x/00-assert.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+apiVersion: kubeaddons.mesosphere.io/v1beta1
+kind: Addon
+metadata:
+  name: generic-application-operator
+
```
Comment (porridge) [inline tests/kubeaddons-enterprise/addons/generic-application-operator/0.x/generic-operator-install-0-x/00-assert.yaml:5]: Shouldn't we also assert something about the state?

File: tests/kubeaddons-enterprise/addons/generic-application-operator/0.x/generic-operator-install-0-x/01-update.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+kind: Addon
+metadata:
+  name: generic-application-operator
+spec:
+  kudoReference:
+    parameters: |
+      APPLICATION_NAME: nginx
+      PORT: 80
+      ENV: '{"foo": "bar", "bam": "baz"}'
+      IMAGE: nginx
```
Comment (porridge) [inline tests/kubeaddons-enterprise/addons/generic-application-operator/0.x/generic-operator-install-0-x/01-update.yaml:11]: Should we assert that at least some parameters are propagated?

**Review by porridge — APPROVED** [review]

File: tests/kubeaddons-enterprise/addons/generic-application-operator/0.x/generic-operator-install-0-x/00-assert.yaml (hunk 2 above)
Comment (zmalik) [reply tests/kubeaddons-enterprise/addons/generic-application-operator/0.x/generic-operator-install-0-x/00-assert.yaml:5]: the addon installation of the generic operator itself doesn't create anything

File: tests/kubeaddons-enterprise/addons/generic-application-operator/0.x/generic-operator-install-0-x/01-update.yaml (hunk 3 above)
Comment (zmalik) [reply tests/kubeaddons-enterprise/addons/generic-application-operator/0.x/generic-operator-install-0-x/01-update.yaml:11]: we don't assert parameters but the nginx deployment is created using these parameters. If application name was `my-app` then assert would for a deployment `generic-application-operator-my-app`

File: tests/kubeaddons-enterprise/addons/generic-application-operator/0.x/generic-operator-install-0-x/00-install.yaml (hunk 1 above)
Comment (zmalik) [reply tests/kubeaddons-enterprise/addons/generic-application-operator/0.x/generic-operator-install-0-x/00-install.yaml:4]: we could actually do it but using env variable felt easier as these tests are called from `kubeaddons-enterprise` repo CI step also

**Pull request merged**